File: app/src/main/java/com/rubyhuntersky/peregrine/TradeDialogFragment.java
package com.rubyhuntersky.peregrine;

import android.support.v7.app.AppCompatDialogFragment;

import java.math.BigDecimal;

/**
 * @author wehjin
 * @since 2/4/16.
 */

public class TradeDialogFragment extends AppCompatDialogFragment {

    public static String getSharesString(BigDecimal shareCount) {
        final String shareCountString = shareCount.setScale(0, BigDecimal.ROUND_HALF_UP).toPlainString();
        return String.format("%s shares", shareCountString);
    }
}

File: app/src/main/java/com/rubyhuntersky/peregrine/ui/TradeDialogFragment.kt
package com.rubyhuntersky.peregrine.ui

import android.support.v7.app.AppCompatDialogFragment

import java.math.BigDecimal

/**
 * @author wehjin
 * *
 * @since 2/4/16.
 */

open class TradeDialogFragment : AppCompatDialogFragment() {
    companion object {
        val BigDecimal.intString: String get() = "${setScale(0, BigDecimal.ROUND_HALF_DOWN).toPlainString()}"
        val BigDecimal.shares: String get() = "${setScale(0, BigDecimal.ROUND_HALF_DOWN).toPlainString()} shares"

        fun getSharesString(shareCount: BigDecimal): String {
            return shareCount.shares
        }
    }
}
